File: JavaScript/1367-Linked List in Binary Tree.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {ListNode} head
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSubPath = function(head, root) {
    let queue = [];
    queue.push(root);\
    
    const search = ( r, h ) => {
        if (h === null) return true;
        if (r === null) return false;
        if (h.val === r.val && (search(r.left, h.next) || search(r.right, h.next))){
            return true;
        } else{
            return false;
        }
    }
    
    while (queue.length > 0){
        let r = queue.shift();
        if (search(r, head) === true) return true;
        if (r.left !== null) queue.push(r.left);
        if (r.right !== null) queue.push(r.right);
    }
    return false;
};

File: C++/1367-Linked List in Binary Tree.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool traversal(ListNode* listHead, TreeNode* curr) {
        if (listHead->next == nullptr) return true;
        if (curr->left == nullptr && curr->right == nullptr) return false;
        
        bool res = false;
        
        if (curr->left != nullptr && listHead->next != nullptr && curr->left->val == listHead->next->val) {
            res |= traversal(listHead->next, curr->left);
        }
        if (curr->right != nullptr && listHead->next != nullptr && curr->right->val == listHead->next->val) {
            res |= traversal(listHead->next, curr->right);
        }

        return res;
    }
    bool dfs(ListNode* listHead, TreeNode* curr) {
        if (curr == nullptr) return false;
        if (curr->val == listHead->val && traversal(listHead, curr)) {
            return true;
        }
        return (dfs(listHead, curr->left) || dfs(listHead, curr->right));
    }
    bool isSubPath(ListNode* head, TreeNode* root) {
        return dfs(head, root);
    }
};